tests/components/google/test_init.py

```diff
@@ -1,67 +1,513 @@
+    mock_code_flow: Mock,
+    mock_exchange: Mock,
+    mock_notification: Mock,
+    mock_load_platform: Mock,
+    calendars_yaml: None,
+    component_setup: ComponentSetup,
+) -> None:
+    """Test successful creds setup."""
+    assert await component_setup()
+
+    # Run one tick to invoke the credential exchange check
+    now = utcnow()
+    await fire_alarm(hass, now + CODE_CHECK_ALARM_TIMEDELTA)
+
+    mock_load_platform.assert_called_once()
+
+    # One calendar configured
+    assert len(hass.data[google.DATA_INDEX]) == 1
+
+    mock_notification.assert_called()
+    assert "We are all setup now" in mock_notification.call_args[0][1]
+
+
+async def test_code_error(
+    hass: HomeAssistant,
+    mock_code_flow: Mock,
+    component_setup: ComponentSetup,
+    mock_notification: Mock,
+) -> None:
+    """Test loading the integration with no existing credentials."""
+
+    with patch(
+        "oauth2client.client.OAuth2WebServerFlow.step1_get_device_and_user_codes",
+        side_effect=OAuth2DeviceCodeError("Test Failure"),
+    ):
+        assert await component_setup()
+
+    mock_notification.assert_called()
+    assert "Error: Test Failure" in mock_notification.call_args[0][1]
+
+
+@pytest.mark.parametrize("code_expiration_delta", [datetime.timedelta(minutes=-5)])
+async def test_expired_after_exchange(
+    hass: HomeAssistant,
+    mock_code_flow: Mock,
+    component_setup: ComponentSetup,
+    mock_notification: Mock,
+) -> None:
+    """Test loading the integration with no existing credentials."""
+
+    assert await component_setup()
+
+    now = utcnow()
+    await fire_alarm(hass, now + CODE_CHECK_ALARM_TIMEDELTA)
+
+    mock_notification.assert_called()
+    assert (
+        "Authentication code expired, please restart Home-Assistant and try again"
+        in mock_notification.call_args[0][1]
+    )
+
+
+async def test_exchange_error(
+    hass: HomeAssistant,
+    mock_code_flow: Mock,
+    component_setup: ComponentSetup,
+    mock_notification: Mock,
+) -> None:
+    """Test an error while exchanging the code for credentials."""
+
+    with patch(
+        "oauth2client.client.OAuth2WebServerFlow.step2_exchange",
+        side_effect=FlowExchangeError(),
+    ):
+        assert await component_setup()
+
+        now = utcnow()
+        await fire_alarm(hass, now + CODE_CHECK_ALARM_TIMEDELTA)
+
+    mock_notification.assert_called()
+    assert "In order to authorize Home-Assistant" in mock_notification.call_args[0][1]
+
+
+async def test_existing_token(
+    hass: HomeAssistant,
+    token_file: None,
+    component_setup: ComponentSetup,
+    mock_load_platform: Mock,
+    google_service: GoogleCalendarService,
+    calendars_yaml: None,
+    mock_notification: Mock,
+) -> None:
+    """Test setup with an existing token file."""
+    assert await component_setup()
+
+    mock_load_platform.assert_called_once()
+
+    # One calendar configured
+    assert len(hass.data[google.DATA_INDEX]) == 1
```

Comment:
We shouldn't assert any details like hass.data. We can check that there's a state in the state machine instead.

Reply:
Fixed, now looks at entities.